src/components/bookmark-list-container.ts

```diff
@@ -142,95 +80,13 @@ export class BookmarkListContainer extends LitElement {
     }
   }
 
-  private checkOngoingSync() {
-    if (SyncService.isSyncInProgress()) {
-      const progress = SyncService.getCurrentSyncProgress();
-      this.containerState = {
-        ...this.containerState,
-        isSyncing: true,
-        syncProgress: progress.current,
-        syncTotal: progress.total,
-        syncedBookmarkIds: new Set<number>(),
-      };
-    }
-  }
-
-  private initializeFavicons() {
-    // Initialize empty favicon cache - will be populated on demand
-    this.containerState = {
-      ...this.containerState,
-      faviconCache: new Map<number, string>(),
-    };
-  }
-
-  // Event handlers
-  private handleSyncInitiated = () => {
-    // Show immediate feedback with indeterminate progress
-    this.containerState = {
-      ...this.containerState,
-      isSyncing: true,
-      syncProgress: 0,
-      syncTotal: 0,
-      syncedBookmarkIds: new Set<number>(),
-    };
-  };
-
-  private handleSyncStarted = (event: Event) => {
-    const customEvent = event as CustomEvent;
-    this.containerState = {
-      ...this.containerState,
-      isSyncing: true,
-      syncProgress: 0,
-      syncTotal: customEvent.detail.total || 0,
-      syncedBookmarkIds: new Set<number>(),
-    };
-  };
-
-  private handleSyncProgress = (event: Event) => {
-    const customEvent = event as CustomEvent;
-    this.containerState = {
-      ...this.containerState,
-      syncProgress: customEvent.detail.current || customEvent.detail.progress || 0,
-      syncTotal: customEvent.detail.total || 0,
-    };
-  };
-
-  private handleSyncCompleted = async () => {
-    this.containerState = {
-      ...this.containerState,
-      isSyncing: false,
-      syncProgress: 0,
-      syncTotal: 0,
-    };
-
+  // Controller event handlers
+  private async handleSyncCompleted() {
     // Reload bookmarks after sync
     await this.loadBookmarks();
```

Comment:
Why do we need to reload bookmarks after sync, if the `handleBookmarkSynced()` method is already incrementally loading each bookmark as it is synced?